listenbrainz/timescale_writer/timescale_writer.py

```diff
@@ -51,29 +52,45 @@ def get_consumers(self, _, channel):
         ]
 
     def callback(self, message: Message):
-        listens = orjson.loads(message.body)
+        """ Process a message from the incoming queue.
+
+            1. If a listen inserts successfully, send it to the unique queue. Ack the incoming message.
+            2. If a listen fails to insert, send it to the rejection queue. Ack the incoming message. If
+               adding the message to the rejection queue fails, bubble up the error to cause a service
+               restart.
+            3. If a listen fails to insert due to a database error, sleep for a few seconds and bubble up
+               the error to cause a service restart.
+        """
+        try:
+            listens = orjson.loads(message.body)
 
-        msb_listens = []
-        for chunk in chunked(listens, MAX_ITEMS_PER_MESSYBRAINZ_LOOKUP):
-            msb_listens.extend(self.messybrainz_lookup(chunk))
+            msb_listens = []
+            for chunk in chunked(listens, MAX_ITEMS_PER_MESSYBRAINZ_LOOKUP):
+                msb_listens.extend(self.messybrainz_lookup(chunk))
 
-        submit = []
-        for listen in msb_listens:
+            submit = [Listen.from_json(listen) for listen in msb_listens]
+            self.insert_to_listenstore(submit)
+        except psycopg2.OperationalError as e:
+            current_app.logger.error("Error processing listens due to database issues: %s" % str(e), exc_info=True)
+            time.sleep(self.ERROR_RETRY_DELAY)
+            raise
+        except Exception as e:
+            current_app.logger.error("Error processing listens, publishing to rejection queue: %s" % str(e), exc_info=True)
```

Comment:
From what I understand, `exc_info=True` already includes the error in the logger, no? Wouldn't this duplicate the error logging?
```suggestion
            current_app.logger.error("Error processing listens, publishing to rejection queue:", exc_info=True)
```